src/main/java/me/zeroeightsix/kami/module/modules/movement/HoleSnap.kt

```diff
@@ -0,0 +1,67 @@
+package me.zeroeightsix.kami.module.modules.movement
+
+import me.zeroeightsix.kami.module.Module
+import me.zeroeightsix.kami.setting.Settings
+import me.zeroeightsix.kami.util.combat.SurroundUtils
+import me.zeroeightsix.kami.util.math.RotationUtils
+import me.zeroeightsix.kami.util.math.VectorUtils
+import me.zeroeightsix.kami.util.math.VectorUtils.toBlockPos
+import me.zeroeightsix.kami.util.math.VectorUtils.toVec3d
+import net.minecraft.util.math.BlockPos
+import kotlin.math.*
+
+@Module.Info(
+        name = "HoleSnap",
+        description = "Move you into the hole nearby",
+        category = Module.Category.MOVEMENT
+)
+object HoleSnap : Module() {
+    private val disableStrafe = register(Settings.b("DisableStrafe",true))
+    private val range = register(Settings.floatBuilder("Range").withValue(2.5f).withRange(0f, 5f))
+
+    override fun onEnable() {
+        if (mc.player == null) disable()
+    }
+
+    override fun onDisable() {
+        mc.timer.tickLength = 50f
+    }
+
+    override fun onUpdate() {
+        if (SurroundUtils.checkHole(mc.player) != SurroundUtils.HoleType.NONE) {
+            disable()
+            return
+        }
+        findHole()?.toVec3d()?.let {
+            if (disableStrafe.value) Strafe.disable()
```

Comment:
see above

Reply:
It disables after you got into a hole. There is no reason to enabled strafe in hole